builtin/providers/aws/provider.go

```diff
@@ -10,10 +10,15 @@ import (
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/terraform"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
 )
 
+var defaultClient = ec2metadata.New(session.New(aws.NewConfig()))
```

Comment:
The additional `aws.NewConfig()` is not required when creating a new session. You could do just, `session.New()`. The parameter New takes is an optional variadic list of configs which will be merged on top of the session's copy of the default config sourced from the SDK's `defaults` package.